src/SourceLocator/SourceStubber/PhpStormStubsSourceStubber.php

```diff
@@ -0,0 +1,227 @@
+                $this->className = $className;
+            }
+
+            public function enterNode(Node $node) : ?int
+            {
+                if ($node instanceof Node\Stmt\Namespace_) {
+                    return null;
+                }
+
+                if ($node instanceof Node\Stmt\ClassLike && $node->namespacedName->toString() === $this->className) {
+                    $this->node = $node;
+                    return NodeTraverser::STOP_TRAVERSAL;
+                }
+
+                return NodeTraverser::DONT_TRAVERSE_CHILDREN;
+            }
+
+            public function getNode() : ?Node\Stmt\ClassLike
+            {
+                return $this->node;
+            }
+        };
+    }
+
+    private function getFunctionNodeVisitor(CoreReflectionFunction $functionReflection) : NodeVisitorAbstract
+    {
+        return new class($functionReflection->getName()) extends NodeVisitorAbstract
+        {
+            /** @var string */
+            private $functionName;
+
+            /** @var Node\Stmt\Function_|null */
+            private $node;
+
+            public function __construct(string $className)
+            {
+                $this->functionName = $className;
+            }
+
+            public function enterNode(Node $node) : ?int
+            {
+                if ($node instanceof Node\Stmt\Namespace_) {
+                    return null;
+                }
+
+                /** @psalm-suppress UndefinedPropertyFetch */
+                if ($node instanceof Node\Stmt\Function_ && $node->namespacedName->toString() === $this->functionName) {
+                    $this->node = $node;
+                    return NodeTraverser::STOP_TRAVERSAL;
+                }
+
+                return NodeTraverser::DONT_TRAVERSE_CHILDREN;
+            }
+
+            public function getNode() : ?Node\Stmt\Function_
+            {
+                return $this->node;
+            }
+        };
+    }
+
+    /**
+     * @return string[]
+     */
+    private function getExtensionStubsFiles(string $extensionName) : array
+    {
+        if (array_key_exists($extensionName, $this->extensionStubsFiles)) {
+            return $this->extensionStubsFiles[$extensionName];
+        }
+
+        $this->extensionStubsFiles[$extensionName] = [];
+
+        $extensionDirectory = sprintf('%s/%s', $this->getStubsDirectory(), $extensionName);
+
+        if (! is_dir($extensionDirectory)) {
+            return [];
+        }
+
+        foreach (new DirectoryIterator($extensionDirectory) as $fileInfo) {
+            if ($fileInfo->isDot()) {
+                continue;
+            }
+
+            $this->extensionStubsFiles[$extensionName][] = $fileInfo->getPathname();
+        }
+
+        return $this->extensionStubsFiles[$extensionName];
+    }
+
+    private function getStubsDirectory() : string
+    {
+        if ($this->stubsDirectory !== null) {
+            return $this->stubsDirectory;
+        }
+
+        $directories = [
```

Comment:
These should be moved to a `private const`, for easier lookup at the top of this file